### PR TITLE
Update readme to include theme instructions

### PR DESCRIPTION
**Pull request by mattmc3**
Per the convo in #708, here's an update to the readme with two new sections regarding theme support.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,36 @@ A plugin can be any number of files in a `functions`, `conf.d`, and `completions
 
 Non `.fish` files as well as directories inside those locations will be copied to `$fisher_path` under `functions`, `conf.d`, or `completions` respectively.
 
+## Creating a theme
+
+A theme is just like a regular Fish plugin, except it has a `.theme` file in the `themes` directory. Themes were introduced in [Fish 3.4](https://github.com/fish-shell/fish-shell/releases/tag/3.4.0) and work with the `fish_config` builtin. A theme can also have files in `functions`, `conf.d`, or `completions` if necessary. This is what a typical theme plugin might look like.
+
+<pre>
+<b>sosuke</b>
```
Comment (jorgebucaran) [inline README.md:136]: 😂

**Comment by jorgebucaran** [comment]
Thank you, @mattmc3! 💯

**Comment by PatrickF1** [comment]
The organization of the readme is now off. "Event system" and "Discoverability" are now nested under "Creating a theme" even though they are orthogonal concerns to themes.

**Comment by jorgebucaran** [comment]
@PatrickF1 https://github.com/jorgebucaran/fisher/commit/41996574959b9cce51988a4308a5e57841398d03 What do you think?

**Comment by PatrickF1** [comment]
Great!

**Pull request merged**